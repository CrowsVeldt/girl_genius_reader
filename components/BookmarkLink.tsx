import { useContext, useEffect } from "react";
import { Text, TouchableOpacity, View } from "react-native";
import { StyleSheet } from "react-native";
import { DateContext } from "../context/DateContext";
import { formatDate } from "../utils/utilFunctions";

export default function BookmarkLink({
  date,
  nav,
}: {
  date: string;
  nav: any;
}) {
  const { removeBookmark, changeCurrentDate } = useContext(DateContext);

  useEffect(() => {});
  return (
    <View style={styles.bookmark}>
      <TouchableOpacity
        style={styles.linkButton}
        onPress={() => {
          changeCurrentDate(date);
          nav.navigate("Home");
        }}
      >
        <Text style={styles.linkText}>{formatDate(date)}</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.removeButton}
        onPress={() => removeBookmark(date)}
      >
        <Text>X</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  bookmark: {
    flexDirection: "row",
    justifyContent: "space-between",
    borderWidth: 1,
    height: 35,
  },
  removeButton: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 1,
    width: 30,
  },
  linkButton: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    flexGrow: 1,
    borderWidth: 1,
  },
  linkText: {},
});
