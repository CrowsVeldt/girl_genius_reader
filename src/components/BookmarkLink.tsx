import { useContext } from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import { ComicContext } from "../context/ComicContext";
import { PageType } from "../utils/types";

export default function BookmarkLink({
  page,
  nav,
}: {
  page: PageType;
  nav: any;
}) {
  const { removeBookmark, changeCurrentPage } = useContext(ComicContext);

  return (
    <View style={styles.bookmark}>
      <TouchableOpacity
        style={styles.linkButton}
        onPress={() => {
          changeCurrentPage(page);
          nav.navigate("Home");
        }}
      >
        <Text>{`Volume ${page.volumeNumber} - Page ${page.pageNumber}`}</Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.removeButton}
        onPress={() => removeBookmark(page)}
      >
        <Text>X</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  bookmark: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignSelf: "center",
    borderWidth: 1,
    height: 35,
    width: "100%",
  },
  removeButton: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 1,
    width: 30,
  },
  linkButton: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    flexGrow: 1,
    borderWidth: 1,
  },
});
