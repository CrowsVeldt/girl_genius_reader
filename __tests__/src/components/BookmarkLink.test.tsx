import renderer from "react-test-renderer";
import BookmarkLink from "../../../src/components/BookmarkLink";
import ComicProvider from "../../../src/context/ComicContext";

describe("<BookmarkLink />", () => {
  const testPage = {
    date: "20021104",
    title: "",
    pageNumber: 1,
    volume: 1,
  };

  it("renders the same as last time", () => {
    const tree = renderer.create(
      <ComicProvider>
        <BookmarkLink page={testPage} nav={jest.mock} />
      </ComicProvider>
    );
    expect(tree).toMatchSnapshot();
  });
});
