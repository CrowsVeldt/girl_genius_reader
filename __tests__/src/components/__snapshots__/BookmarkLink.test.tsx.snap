// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookmarkLink /> renders the same as last time 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 35,
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderWidth": 1,
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text>
      Volume 1 - Page 1
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "width": 30,
      }
    }
  >
    <Text>
      X
    </Text>
  </View>
</View>
`;
